azavea/raster-foundry
# Refactor map container

## Overview

Refactors map interaction to remove some of the spaghetti code that resulted from everything being controlled through watched props.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~

### Notes

I think that eventually, we'll want to make the geojson stuff an extension on top of the leaflet geojson layer, instead of a part of the wapper. That way, there's no difference between adding a normal layer and adding geojson.

## Testing Instructions

* On the prowse page, verify that:
  * The scene grid displays correctly, with mouseover counts, and refreshes whenever the scene list does
  * Mousing over a scene shows its footprint on the map
  * Filters work
  * Zooming / panning updates the bbox url parameter and updates the scene list
* On Scene Detail / Project Scene Detail page:
  * Shows the footprint of the scene
  * Map view is fit to footprint polygon
  * All interaction disabled
* On the Project editor views:
  * Mouseover of scenes in both color correction and mosiac shows footprint on map
  * Loading the page in either subview loads scenes / scene layers
  * Switching subviews does not cause re-loading scenes
  * Clicking a grid cell in the color correction view selects the grid cell & all scenes inside it, and highlights the selected grid cell
  * Color correction still functions
  * Scene focus still function
  * Shape creation / editing / deletion works in the mosaic mask view
  * Re-ordered scenes stay ordered in the mosaic / color correction view
